Modules/Devices/test/Devices.Application.Tests/Tests/Identities/Commands/CancelDeletionProcess/HandlerTests.cs

```diff
@@ -0,0 +1,72 @@
+using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Devices.Domain.Entities.Identities;
+using Backbone.Tooling;
+using FakeItEasy;
+using FluentAssertions;
+using Xunit;
+using static Backbone.UnitTestTools.Data.TestDataGenerator;
+
+namespace Backbone.Modules.Devices.Application.Tests.Tests.Identities.Commands.CancelDeletionProcess;
+public class HandlerTests
+{
+    [Fact]
+    public async Task Happy_path()
+    {
+        // Arrange
+        var activeIdentity = TestDataGenerator.CreateIdentityWithApprovedDeletionProcess(SystemTime.UtcNow.Date);
+        var activeDevice = activeIdentity.Devices[0];
+        var deletionProcess = activeIdentity.GetDeletionProcessInStatus(DeletionProcessStatus.Approved)!;
+
+        var fakeIdentitiesRepository = A.Fake<IIdentitiesRepository>();
+        var fakeUserContext = A.Fake<IUserContext>();
+
+        A.CallTo(() => fakeIdentitiesRepository.FindByAddress(activeIdentity.Address, CancellationToken.None, A<bool>._))
+            .Returns(activeIdentity);
+        A.CallTo(() => fakeUserContext.GetAddress()).Returns(activeIdentity.Address);
+        A.CallTo(() => fakeUserContext.GetDeviceId()).Returns(activeDevice.Id);
+
+        var handler = CreateHandler(fakeUserContext, fakeIdentitiesRepository);
+        var command = new CancelDeletionProcessCommand(deletionProcess.Id);
+
+        // Act
+        var response = await handler.Handle(command, CancellationToken.None);
+
+        // Assert
+        A.CallTo(() => fakeIdentitiesRepository.Update(A<Identity>.That.Matches(i =>
+                i.Address == activeIdentity.Address
+                && i.Status == IdentityStatus.Active
+                && i.DeletionProcesses.Any(d => d.Id == deletionProcess.Id)), A<CancellationToken>._))
+            .MustHaveHappenedOnceExactly();
+
+        response.Status.Should().Be(DeletionProcessStatus.Cancelled);
+    }
+
+    [Fact]
+    public void Cannot_start_when_given_identity_does_not_exist()
+    {
+        // Arrange
+        var address = CreateRandomIdentityAddress();
+        var fakeIdentitiesRepository = A.Fake<IIdentitiesRepository>();
+        var fakeUserContext = A.Fake<IUserContext>();
+
+        A.CallTo(() => fakeIdentitiesRepository.FindByAddress(address, CancellationToken.None, A<bool>._)).Returns<Identity>(null);
+
+        var handler = CreateHandler(fakeUserContext);
+
+        // Act
+        var acting = async () => await handler.Handle(new CancelDeletionProcessCommand(address), CancellationToken.None);
+
+        // Assert
+        acting.Should().ThrowAsync<NotFoundException>();
+    }
+
+    private static Handler CreateHandler(IUserContext userContext, IIdentitiesRepository? identityRepository = null)
+    {
+        return identityRepository == null
+            ? new Handler(A.Fake<IIdentitiesRepository>(), userContext)
+            : new Handler(identityRepository, userContext);
```

Comment:
Why so complicated? All the other `CreateHandler` methods look different:

```suggestion
        identitiesRepository ??= A.Fake<IIdentitiesRepository>()
        return new Handler(identityRepository, userContext);
```

But keep in mind that according to the comment above, the `userContext` parameter should become optional, and the `identitiesRepository` should become mandatory.

Reply:
Oh, damn, now I see how the `CreateHandler` methods look like in other tests. I should have taken a look into them. My bad. Also, I fixed which parameter is optional and which is mandatory.